examples/reaction_roles.py

```diff
@@ -0,0 +1,34 @@
+"""Uses a messages to add and remove roles through reactions."""
+
+import discord
+from discord.ext import commands
+
+# This bot requires the members and reactions intensions.
+intents = discord.Intents.default()
+intents.members = True
+bot = commands.Bot(command_prefix=commands.when_mentioned_or("!"), intents=intents)
+
+role_message_id = 0  # ID of message that can be reacted to to add role
+emoji_to_role = {
+    "👍": 0,  # ID of role associated with thumbs up emoji
+    "test": 0  # ID of role associated with custom emoji 'test'
+}
+
+@bot.event
+async def on_raw_reaction_add(payload):
+    """Gives a role based on a reaction emoji."""
+    if payload.message_id == role_message_id and payload.emoji.name in emoji_to_role:
+        guild = bot.get_guild(payload.guild_id)
+        role = guild.get_role(emoji_to_role[payload.emoji.name])
+        await payload.member.add_roles(role)
```

Comment:
```suggestion
    # Make sure that the message the user is reacting to is the one we care about
    if payload.message_id != role_message_id:
        return
    
    try:
        role_id = emoji_to_role[str(payload.emoji)]
    except KeyError:
        # If the emoji isn't the one we care about then exit as well.
        return
    
    guild = bot.get_guild(payload.guild_id)
    if guild is None:
        # Check if we're still in the guild and it's cached.
        return

    role = guild.get_role(role_id)
    if role is None:
        # Make sure the role still exists and is valid.
        return

    try:
        # Finally add the role
        await payload.member.add_roles(role)
    except discord.HTTPException:
        # If we want to do something in case of errors we'd do it here.
        pass
```

Generally this is the style I would consider good for the examples. I left the other pair as an exercise.